examples/02_decoding/plot_miyawaki_encoding.py

```diff
@@ -0,0 +1,261 @@
+plt.subplots_adjust(wspace=0.5)
+
+##############################################################################
+# We now stack the fmri and stimulus data and remove an offset in the
+# beginning/end.
+
+fmri_data = np.vstack([fmri_run[2:] for fmri_run in fmri_data])
+stimuli = np.vstack([stimuli_run[:-2] for stimuli_run in stimuli]).astype(float)
+
+##############################################################################
+# fmri_data is a matrix of *samples* x *voxels*
+
+print(fmri_data.shape)
+
+##############################################################################
+# We flatten the last two dimensions of stimuli
+# so it is a matrix of *samples* x *pixels*.
+
+# Flatten the stimuli
+stimuli = np.reshape(stimuli, (-1, stimulus_shape[0] * stimulus_shape[1]))
+
+print(stimuli.shape)
+
+##############################################################################
+# Building the encoding models
+# ----------------------------
+# We can now proceed to build a simple **voxel-wise encoding model** using
+# `Ridge regression <http://en.wikipedia.org/wiki/Tikhonov_regularization>`_.
+# For each voxel we fit an independent regression model,
+# using the pixel-values of the visual stimuli to predict the neuronal
+# activity in this voxel.
+
+from sklearn.linear_model import Ridge
+from sklearn.cross_validation import KFold
+
+##############################################################################
+# Using 10-fold cross-validation, we partition the data into 10 'folds'.
+# We hold out each fold of the data for testing, then fit a ridge regression
+# to the remaining 9/10 of the data, using stimuli as predictors
+# and fmri_data as targets, and create predictions for the held-out 10th.
+from sklearn.metrics import r2_score
+
+estimator = Ridge(alpha=100.)
+cv = KFold(len(stimuli), 10)
+
+scores = []
+for train, test in cv:
+    # we train the Ridge estimator on the training set
+    # and predict the fMRI activity for the test set
+    predictions = Ridge(alpha=100.).fit(
+    stimuli.reshape(-1, 100)[train], fmri_data[train]).predict(
+        stimuli.reshape(-1, 100)[test])
+    # we compute how much variance our encoding model explains in each voxel
+    scores.append(r2_score(fmri_data[test], predictions,
+                           multioutput='raw_values'))
+
+##############################################################################
+# Mapping the encoding scores on the brain
+# ----------------------------------------
+# To plot the scores onto the brain, we use :func:`nilearn.image.new_img_like`
+# to create a Nifti1Image containing the scores and then threshold it:
+
+from nilearn.image import threshold_img, new_img_like
+
+cut_score = np.mean(scores, axis=0)
+cut_score[cut_score < 0] = 0
+
+# bring the scores into the shape of the background brain
+score_map_img = new_img_like(dataset.background,
+                             masker.inverse_transform(cut_score).get_data())
```

Comment:
> The mask doesn't have the affine transform of the background image.

Hum... One of the two must be wrong, then. Don't you think? I can't
really see another explaination.